### PR TITLE
Feature/sl 892 e2e single rules engine

### PR DESCRIPTION
**Pull request by kpakur**
https://tools.hmcts.net/jira/browse/SL-892

Implemented returning facts created in the response so it may be used by the service submitting facts to the rules engine.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlrules/services/DroolsService.java [hunk 1]
```diff
@@ -30,11 +36,21 @@ public void init() {
         KieContainer kieContainer = kieServices.getKieClasspathContainer();
         rulesSession = kieContainer.newKieSession(droolsConfiguration.getRulesKSession());
 
-        rulesSession.addEventListener(new FactsChangedEventListener());
-        rulesSession.addEventListener(new RulesChangedEventListener());
+        factsChangedEventListener = new FactsChangedEventListener();
+
+        rulesSession.addEventListener(factsChangedEventListener);
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlrules/services/DroolsService.java]: Could you move instantiation of FactsChangedEventListener to addEventListener and also remove private field?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlrules/drools/FactModification.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.Data;
+import uk.gov.hmcts.reform.sandl.snlrules.model.Fact;
+
+@Data
+public class FactModification {
+    private Fact oldFact;
+    private Fact newFact;
+
+    public FactModification(Fact oldFact, Fact newFact) {
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlrules/drools/FactModification.java]: Is this needed? @AllArgsConstructor should work?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlrules/services/DroolsService.java (hunk 1 above)
Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlrules/services/DroolsService.java]: no :) it's used in methods below

File: src/main/java/uk/gov/hmcts/reform/sandl/snlrules/drools/FactModification.java (hunk 2 above)
Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlrules/drools/FactModification.java]: true, not needed

**Pull request merged**